### PR TITLE
examples + drivername

### LINKED ISSUE #38: Add domain to driver name
Maybe hostpath.kubernetes-csi.io?

### PR DESCRIPTION
**Pull request by pohly**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:

We want the name to follow the recommendation from the spec and use a domain. While testing this, some issues with the examples were noticed and fixed.

**Which issue(s) this PR fixes**:

Fixes #38

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
the deployment uses `hostpath.csi.k8s.io` as driver name
```

### CODE REVIEW

**Comment by pohly** [comment]
Looks like I need to update csi-release-tools first with a more flexible way of creating the external driver test config. Hard-coding `csi-hostpath` as name no longer works.

/hold

**Comment by pohly** [comment]
@msau42: we previously postponed updating Go to 1.12 (PR #39). Because this PR here updates csi-release-tools, it also includes that change. I lost track of how long we wanted to postpone, though. Probably it was something that should have been in csi-driver-host-path for Kubernetes 1.15?

Anyway, this PR is becoming more complicated than expected. If you want to include just the examples fix, then merge PR #65 instead.

**Comment by msau42** [comment]
Yes we should update go version now

File: README.md [hunk 1]
```diff
@@ -82,7 +82,7 @@ csi-hostpathplugin-0         2/2     Running   0          5m45s
 From the root directory, deploy the application pods including a storage class, a PVC, and a pod which mounts a volume using the Hostpath driver found in directory `./examples`:
 
 ```shell
-$ kubectl create -f ./examples
+$ for i in ./examples/csi-storageclass.yaml ./examples/csi-pvc.yaml ./examples/csi-app.yaml; do kubectl apply -f $i; done
```
Comment (msau42) [inline README.md]: should we explicitly call out each file as a separate step?

**Comment by msau42** [comment]
Is there a way to get rid of this merge commit: 6128d3d61e6f9d598e164bef5687e6df637ee716?

**Comment by pohly** [comment]
> Is there a way to get rid of this merge commit: 6128d3d?

That's the `release-tools` update, which is needed for changing the driver name, and importing with `git subtree` has to be done with merge commits that splices together the external repo and the one it is getting merged into. So, no, this cannot be avoided. I can change the commit message, if that helps.

File: README.md (hunk 1 above)
Comment (pohly) [reply README.md]: We could. It would be more work to copy-and-paste the whole sequence. I don't have a strong opinion here.

**Pull request merged**